src/backtracking/knight_tour.rs

```diff
@@ -0,0 +1,206 @@
+    fn is_safe(&self, x: i32, y: i32) -> bool {
+        x >= 0
+            && y >= 0
+            && x < self.board_size as i32
+            && y < self.board_size as i32
+            && self.board[x as usize][y as usize] == 0
+    }
+
+    /// Recursively solves the Knight's Tour problem.
+    ///
+    /// # Arguments
+    ///
+    /// * `x` - The current x-coordinate of the knight.
+    /// * `y` - The current y-coordinate of the knight.
+    /// * `move_count` - The current move count.
+    ///
+    /// # Returns
+    ///
+    /// A boolean indicating whether a solution was found.
+    fn solve_tour(&mut self, x: i32, y: i32, move_count: i32) -> bool {
+        if move_count == self.board_size as i32 * self.board_size as i32 {
+            return true;
+        }
+
+        for i in 0..8 {
+            let next_x = x + self.moves[i].0;
+            let next_y = y + self.moves[i].1;
+
+            if self.is_safe(next_x, next_y) {
+                self.board[next_x as usize][next_y as usize] = move_count + 1;
+
+                if self.solve_tour(next_x, next_y, move_count + 1) {
+                    return true;
+                }
+                // Backtrack
+                self.board[next_x as usize][next_y as usize] = 0;
+            }
+        }
+
+        false
+    }
+
+    /// Finds the Knight's Tour starting from the specified position.
+    ///
+    /// # Arguments
+    ///
+    /// * `start_x` - The x-coordinate of the starting position.
+    /// * `start_y` - The y-coordinate of the starting position.
+    ///
+    /// # Returns
+    ///
+    /// A tuple containing a boolean indicating whether a tour was found and the tour matrix if found.
+    pub fn find_tour(&mut self, start_x: usize, start_y: usize) -> (bool, Vec<Vec<i32>>) {
+        if !self.is_safe(start_x as i32, start_y as i32) {
+            println!("Invalid starting position");
+            return (false, vec![]);
+        }
+
+        self.board[start_x][start_y] = 1;
+
+        if !self.solve_tour(start_x as i32, start_y as i32, 1) {
+            println!("No solution exists");
+            return (false, vec![]);
+        }
+
+        println!("Knight's Tour:");
+        (true, self.board.clone())
+    }
+}
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+
+    #[test]
+    fn test_knights_tour_5x5() {
+        let mut tour = KnightTour::new(5);
+        let (found, tour_matrix) = tour.find_tour(0, 0);
+        assert!(found);
+
+        // Define the expected tour matrix for an 8x8 board (assuming a successful tour)
+        let expected_matrix: Vec<Vec<i32>> = vec![
+            vec![1, 6, 15, 10, 21],
+            vec![14, 9, 20, 5, 16],
+            vec![19, 2, 7, 22, 11],
+            vec![8, 13, 24, 17, 4],
+            vec![25, 18, 3, 12, 23],
+        ];
+
+        // Compare each cell of the generated tour matrix with the expected matrix
+        for i in 0..5 {
+            assert_eq!(tour_matrix[i], expected_matrix[i]);
+        }
```

Comment:
What do you think about making these tests more _direst_:
```suggestion
        assert_eq!(tour_matrix, expected_matrix)
```

Reply:
Well, It's will make the code more flexible and readable, especially if dealing with larger matrices. Done! :rocket: